src/mongocxx/options/create_collection.hpp

```diff
@@ -0,0 +1,119 @@
+
+#pragma once
+
+#include <mongocxx/config/prelude.hpp>
+
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/stdx/optional.hpp>
+#include <mongocxx/stdx.hpp>
+
+namespace mongocxx {
+MONGOCXX_INLINE_NAMESPACE_BEGIN
+namespace options {
+
+///
+/// Class representing the optional arguments to a MongoDB createCollection command
+///
+class MONGOCXX_API create_collection {
+   public:
+    ///
+    /// To create a capped collection, specify true.
+    ///
+    /// @note If you specify true, you must also set a maximum size using the size() method.
+    ///
+    /// @param capped
+    ///   Whether or not this collection will be capped.
+    ///
+    /// @see https://docs.mongodb.org/manual/reference/glossary/#term-capped-collection
+    ///
+    void capped(bool capped);
+
+    ///
+    /// Specify false to disable the automatic creation of an index on the _id field.
+    ///
+    /// @note For replica sets, all collections must have autoIndexId set to true.
+    ///
+    /// @param auto_index_id
+    ///   Whether or not this collection will automatically generate an index on _id.
+    ///
+    void auto_index_id(bool auto_index_id);
+
+    ///
+    /// A maximum size, in bytes, for a capped collection.
+    ///
+    /// @note Once a capped collection reaches its maximum size, MongoDB removes older
+    ///   documents to make space for new documents.
+    ///
+    /// @note Size is required for capped collections and ignored for other collections.
+    ///
+    /// @param max_size
+    ///   Maximum size, in bytes, of this collection (if capped)
+    ///
+    void size(int max_size);
+
+    ///
+    /// The maximum number of documents allowed in the capped collection.
+    ///
+    /// @note The size limit takes precedence over this limit. If a capped collection reaches
+    ///   the size limit before it reaches the maximum number of documents, MongoDB removes
+    ///   old documents.
+    ///
+    /// @param max_documents
+    ///   Maximum number of documents allowed in the collection (if capped)
+    ///
+    void max(int max_documents);
+
+    ///
+    /// Specify configuration to the storage on a per-collection basis.
+    ///
+    /// @note This option is currently only available with the WiredTiger storage engine.
+    ///
+    /// @param storage_engine_options
+    ///   Configuration options specific to the storage engine.
+    ///
+    void storage_engine(bsoncxx::document::value storage_engine_opts);
+
+    ///
+    /// When true, disables the power of 2 sizes allocation for the collection.
+    ///
+    /// @see: https://docs.mongodb.org/manual/reference/method/db.createCollection/
+    ///
+    /// @param no_padding
+    ///   When true, disables power of 2 sizing for this collection.
+    ///
+    void no_padding(bool no_padding);
+
+    ///
+    /// Return a bson document representing the options set on this object.
+    ///
+    /// @return Options, as a document.
+    ///
+    bsoncxx::document::value to_document();
```

Comment:
Good point, will provide both.

`value()` is a bit weird because it's not immediately clear that we mean `bsoncxx::document::value` (I mean it is if you go read the header, but won't be from code).  Slight preference for `to_document`